File: python/customer.py
from movie import Movie


class Customer:
    def __init__(self, name):
        self._name = name
        self._rentals = []

    @property
    def name(self):
        return self._name

    def add_rental(self, arg):
        self._rentals.append(arg)

    def statement(self):
        total_amount = 0
        frequent_renter_points = 0
        result = f"Rental Record for {self.name}\n"
        for element in self._rentals:
            frequent_renter_points += element.frequent_renter_points

            # show figures for this rental
            result += "\t" + element.movie.title + "\t" + str(element.charge) + "\n"
            total_amount += element.charge

        # add footer lines
        result += f"Amount owed is {total_amount}\n"
        result += f"You earned {frequent_renter_points} frequent renter points"
        return result

File: ruby/page_013/customer.rb
require 'movie'

class Customer
	attr_reader :name
	def initialize(name)
		@name = name
		@rentals = []
	end
	def add_rental(arg)
		@rentals << arg
	end
	def statement
		total_amount, frequent_renter_points = 0, 0
		result = "Rental Record for #{@name}\n"
		@rentals.each do |element|
			this_amount = amount_for(element)

			# add frequent renter points
			frequent_renter_points += 1
			# add bonus for a two day new release rental
			if element.movie.price_code == Movie::NEW_RELEASE && element.days_rented > 1
				frequent_renter_points += 1
			end

			# show figures for this rental
			result += "\t" + element.movie.title + "\t" + this_amount.to_s + "\n"
			total_amount += this_amount
		end
		# add footer lines
		result += "Amount owed is #{total_amount}\n"
		result += "You earned #{frequent_renter_points} frequent renter points"
		result
	end

    def amount_for(rental)
        rental.charge
    end
end
